labgrid-web-client/src/app/resource-selector/resource-selector.component.ts

```diff
@@ -0,0 +1,156 @@
+  loading = true;
+  resources: Resource[] = [];
+
+  assignedResources: string[] = [];
+  availableResources: string[] = [];
+
+  constructor(
+    private _ps: PlaceService,
+    private _rs: ResourceService,
+    private route: ActivatedRoute,
+    private _router: Router,
+    private _snackBar: MatSnackBar,
+    private _dialog: MatDialog) {
+    this.getPlaceData()
+  }
+
+  ngOnInit(): void {
+  }
+
+  private getResources(): void {
+    this._rs.getResourcesForPlace(this.place.name).then((resources) => {
+      this.resources = resources;
+      // this.assignedResources = resources;
+      resources.forEach(x => {
+        if (x.acquired) {
+          this.assignedResources.push(x.name);
+        } else {
+          this.availableResources.push(x.name);
+        }
+      })
+    }).then(() => this.loading = false);
+
+    /*this._rs.getResourcesForPlace(this.place.name).then((resources) => {
+      this.resources = resources;
+      // this.assignedResources = resources;
+      resources.forEach(x => {
+        this.assignedResources.push(x.name);
+      })
+    });
+
+    this._rs.getResources().then((resources) => {
+      resources.forEach(x => {
+        if (x.acquired === null) {
+          this.availableResources.push(x.name);
+          console.log('Resource: ', x)
+        }
+      })
+    }).then(()=> this.loading = false);*/
+  }
+
+  private getPlaceData() {
+    this.route.params.subscribe((val) => {
+      const currentRoute = this.route.snapshot.url[this.route.snapshot.url.length - 1].path;
+      this._ps.getPlace(currentRoute).then((data) => {
+        // Check if the specified place exists
+        if (data !== undefined) {
+          this.place = data;
+          this.getResources();
+        } else {
+          this._router.navigate(['error']);
+        }
+      });
+    });
+  }
+
+  drop(event: CdkDragDrop<string[]>) {
+    if (event.previousContainer === event.container) {
+      moveItemInArray(event.container.data, event.previousIndex, event.currentIndex);
+    } else {
+      transferArrayItem(
+        event.previousContainer.data,
+        event.container.data,
+        event.previousIndex,
+        event.currentIndex,
+      );
+    }
+  }
+
+  saveResources() {
+    const removedResources: string[] = []
+    const addedResources: string[] = []
+
+    // get deleted resources
+    this.resources.forEach(x => {
+      if (!this.assignedResources.includes(x.name)) {
+        // this resource is not in the assigned List. It has to be removed.
+        removedResources.push(x.name);
+      }
+    })
+
+    // get added resources
+    this.assignedResources.forEach(x => {
+      if (!this.resources.find(y => y.name === x)) {
+        // this resource was added
+        addedResources.push(x);
+      }
+    })
+
+    // console.log('--------results--------')
+    // console.log('removed: ', removedResources)
+    // console.log('added: ', addedResources)
+
+    let response = {successful: false, errorMessage: ''};
```

Comment:
I think you also need to handle the newly acquired resources by calling this._rs.acquireResource().

Reply:
Yes, but at the moment there is just a place that can release resources. As soon as releasing works I will add the acquire functionality.